### PR TITLE
Create build action to upload snapshots

### PR DESCRIPTION
**Pull request by shellderp**
image

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
```
Comment (mpawliszyn) [inline .github/workflows/build.yml:6]: So it releases on every master push?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          npm install -g @misk/cli
+          miskweb ci-build -e
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Set gradle version
+        run: echo "version=String.valueOf(project.version).replace(\"-SNAPSHOT\", \"\") + '-$(date +%Y%m%d)-$(git rev-parse --short HEAD)-SNAPSHOT'" > hooks.gradle
```
Comment (mpawliszyn) [inline .github/workflows/build.yml:32]: But publishes with a date and rev number? I guess this is ok? Do other packages do this?

File: .github/workflows/build.yml (hunk 1 above)
Comment (shellderp) [reply .github/workflows/build.yml:6]: yea

File: .github/workflows/build.yml (hunk 2 above)
Comment (shellderp) [reply .github/workflows/build.yml:32]: SNAPSHOTs get overwritten when you push, so this fixes that problem by having a unique version for each one. Not exactly following conventions but maven conventions suck :p

**Comment by shellderp** [comment]
> Just a couple questions on how happy Sonatype Nexus will be with this

shrug, we don't have that many master pushes anyway. Wire does a snapshot release on every master and seems fine. They don't do the date version thing but I don't see how it's different, the same # of artifacts are stored

**Comment by shellderp** [comment]
I need to look into why there's no actual jar pushed to maven.. wire seems to have the same problem right now. Only sources and docs jars are pushed

**Comment by shellderp** [comment]
> I need to look into why there's no actual jar pushed to maven.. wire seems to have the same problem right now. Only sources and docs jars are pushed

nevermind it works fine

File: .github/workflows/build.yml (hunk 2 above)
Comment (mpawliszyn) [reply .github/workflows/build.yml:32]: SG

**Pull request merged**